### ISSUE TITLE
Extension throws an error when action has invalid payload

### ISSUE THREAD

**Issue by nicocrm** [zalmoxisus/redux-devtools-extension#227]
I (accidentally) had included a DOM node in my action's payload, this caused the extension to crash with a TypeError in _ctx.js.  I tracked it down to this line in the `derez` function:

```
    if (value && typeof value.toJSON === 'function') {
      value = value.toJSON();
    }
```

In my case the toJSON call failed.

While the error was caused by my code, it would be nice to have some error handling so the dev tool could report the problem instead of crashing (the app was working fine without the dev tool open).  Maybe something like this?

```
          try {
        value = value.toJSON();
          } catch (e) {
            value = 'Unable to deserialize: ' + e.toString()
          }
```

I can do a PR but it seems a bit trivial for it.

**Comment by zalmoxisus** [comment]
@nicocrm, thanks for the report. [`jsan`](https://github.com/kolodny/jsan) fails [here](https://github.com/kolodny/jsan/blob/9b37cebc85c5a5cbca0b512764b3a8f5511a93b6/lib/cycle.js#L41) being unable to stringify that object.

Adding the error right in the state subtree value would be helpful, but better to show it explicitly,  to use a `console.error` in your snippet instead (or both of them). Otherwise, it could not be noticed and lead to an unexpected behaviour (for example when exporting / importing the history). 

\cc @kolodny

**Comment by kolodny** [comment]
I'm not sure how to proceed with this. Consider the following:

```js
var a = {toJSON() { throw new Error('test') }};
JSON.stringify({o: a}); // throws error
jsan.stringify({o: a}); // throws error
```
I did notice, however, that I'm calling `toJSON()` without arguments while the native one is called with `(key, value)`. I'll need to fix that.

**Comment by zalmoxisus** [comment]
@nicocrm, I tried to pass a React element and a React component as a state value, and don't seem to get any exceptions.

@kolodny, could you please provide a reference about `key, value` arguments?
I'm calling
```js
JSON.stringify({ foo: 'bar', toJSON: function() { return arguments; } })
```
and don't get any arguments.

**Comment by kolodny** [comment]
You're right, I must have been looking at the replacer function in the docs when I wrote that http://www.ecma-international.org/ecma-262/6.0/#sec-serializejsonproperty

**Comment by nicocrm** [comment]
@zalmoxisus For me the error happened when a native browser event was passed as payload to the action.  I had the action creator as such (because I used a generic function to generate the creators):

    action = x => { type: 'ACTION_TYPE', payload: x }

and in the JSX I had it bound as:

    <button onClick={action} />

*However* I was using Material-UI at the time.  I tried to do a simple repro case without it and was not successful, so I think it might be connected to it.  I'll try and put something together but it might take a little bit because of the dependencies.

I agree that showing it using console.error would be more useful in getting those errors noticed.
